experimental/kubernetes/README.md

```diff
@@ -0,0 +1,186 @@
+  ```
+
+## Deploy Spinnaker Dependencies
+
+1. Copy the Cassandra keyspace scripts to a new directory.
+
+  ```
+  $ gcloud compute ssh root@$MY_GKE_NODE 'mkdir -p /root/cassandra'
+  $ gcloud compute copy-files ../../cassandra root@$MY_GKE_NODE:/root/cassandra
+  ```
+
+1. Deploy the storage mechanisms to the cluster.
+
+  ```
+  $ kubectl create -f 0-dependencies.yml
+  ```
+
+1. Execute each cassandra keyspace script on the cassandra pod.
+
+  ```
+  $ CASS_NAME=`kubectl get pods -l app=cassandra -o go-template='{{ (index .items 0).metadata.name }}'`
+  $ FILES=`ls -1 ../../cassandra/`
+$ for f in $FILES; do \
+    kubectl exec $CASS_NAME -- "cqlsh -f /root/cassandra/$f" \
+  done;
+  ```
+
+1. Enable the Thrift server so that the other Java components can connect to Cassandra
+
+  ```
+  $ kubectl exec $CASS_NAME -- "nodetool enablethrift"
+  ```
+
+## Deploy Component `Services`
+
+1. Deploy the `Service` representation of each Spinnaker component.
+
+  ```
+  $ kubectl create -f 1-services.yml
+  ```
+
+1. After some time, the `deck` and the `gate` services should have external IP addresses. Make note of each of these.
+
+Gate:
+  ```
+  $ kubectl get svc -l component=gate -o go-template='{{ (index (index .items 0).status.loadBalancer.ingress 0).ip }}'
+  ```
+
+Deck:
+  ```
+  $ kubectl get svc -l component=deck -o go-template='{{ (index (index .items 0).status.loadBalancer.ingress 0).ip }}'
+  ```
+
+1. Modify [2-repControllers.yml] to make `gate`'s IP address the value of `deck`'s `API_HOST`
+environmental variable. Look for
+
+```
+- name: API_HOST
+  value: http://GATE_IP_ADDRESS_GOES_HERE:8084
+```
+
+in the configuration.
+
+## Deploy Component `ReplicationController`s
+
+1. Bring up the the actual containers of each component
+
+  ```
+  $ kubectl create -f 2-repControllers.yml
+  ```
+
+1. Monitor that each pod as it comes online
+
+  ```
+  $ kubectl get pods
+  NAME                READY     STATUS    RESTARTS   AGE
+cassandra-szyz9     1/1       Running   0          2d
+clouddriver-8u4i1   1/1       Running   0          27s
+deck-iwkpb          1/1       Running   0          26s
+echo-n7umh          1/1       Running   0          27s
+front50-i4dhl       1/1       Running   0          26s
+gate-dxsti          1/1       Running   0          27s
+orca-7roar          1/1       Running   0          27s
+redis-qqbkq         1/1       Running   0          2d
+rosco-hi2lk         1/1       Running   0          27s
+rush-oy0gx          1/1       Running   0          27s
+```
+
+1. `deck` is one of the slower components to start up. Ensure you see `webpack: bundle is now VALID` at the end of the output log. Check the output with:
+
+  ```
+  $ DECK=`kubectl get pods -l mponent=deck -o go-template='{{ (index .items 0).metadata.name }}'` && kubectl logs -f $DECK
+  ```
+
+1. Access the Spinnaker UI with the Deck external load balancer IP address you acquired earlier: `http://1.2.3.4:9000`
```

Comment:
So does GKE restrict access to ip addresses by default? If not the same disclaimers around ensuring proper firewall rules need to be put here.